### PR TITLE
Feature/traction 289

### PR DESCRIPTION
**Pull request by pserquiz**
# Critical Changes

# Changes
Community user can now submit their status on the home page

# Issues Closed

### CODE REVIEW

File: force-app/main/default/lwc/staffAvailability/staffAvailability.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 <!-- Staff Availability -->
 <template>
+	<c-app-spinner></c-app-spinner>
```
Comment (cwico) [inline force-app/main/default/lwc/staffAvailability/staffAvailability.html:7]: coool

File: force-app/main/default/lwc/staffAvailability/staffAvailability.js [hunk 2]
```diff
@@ -22,8 +23,11 @@ export default class StaffAvailability extends LightningElement {
 	@api statusColorNotAvailable;
 	@api statusColorAssigned;
 
-	namespace;
+	@track staffAvailability;
```
Comment (cwico) [inline force-app/main/default/lwc/staffAvailability/staffAvailability.js]: boolean variable should start with a question, is, has, maybe use isContactAvailable?

File: force-app/main/default/lwc/staffAvailability/staffAvailability.js [hunk 3]
```diff
@@ -79,6 +87,37 @@ export default class StaffAvailability extends LightningElement {
 		}
 	}
 
+	handleToggleStatus(event) {
+		this.staffAvailability = event.target.checked;
```
Comment (cwico) [inline force-app/main/default/lwc/staffAvailability/staffAvailability.js]: this.isContactAvailable makes more sense no?

File: force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js [hunk 4]
```diff
@@ -14,8 +14,8 @@ export default class StaffAvailabilityItem extends LightningElement {
 	ASSIGNED_STATUS = 'Assigned';
 
 	isExpanded = false;
-	statusValue;
-	record;
+	@track statusValue;
```
Comment (cwico) [inline force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js:17]: interesting

File: force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js [hunk 5]
```diff
@@ -35,13 +35,14 @@ export default class StaffAvailabilityItem extends LightningElement {
 		notifyViaAlerts: false // Use alerts instead of toast to notify user
 	};
 
-	renderedCallback() {
-
+	@api
+	get assignmentRecord() {
```
Comment (cwico) [inline force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js:39]: why not null?

File: force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js [hunk 6]
```diff
@@ -35,13 +35,14 @@ export default class StaffAvailabilityItem extends LightningElement {
...
+		return '';
 	}
 
-	connectedCallback() {
-		this.record = {...this.assignment, sobjectType: this.namespace+'Assignment__c'};
-		this.statusValue = getFieldValue(this.record, 'Available__c', this.namespace);
+	set assignmentRecord(value) {
```
Comment (cwico) [inline force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js:43]: this one binds the markup attributre? this is sooo cool

File: force-app/main/default/lwc/staffAvailability/staffAvailability.html [hunk 7]
```diff
@@ -29,18 +30,54 @@ <h2 class="slds-m-right_small">ERROR: </h2>
...
 		</div>
 
 		<div class="slds-card cmp-availability-card">
-			<template if:true={data}>
+			<template if:true={staffAvailability}>
 				<ul class="cmp-grid">
 					<template for:each={data.assignments} for:item="assignment" for:index="index">
 						<li key={assignment} class="slds-border_bottom">
-							<c-staff-availability-item namespace={namespace} assignment={assignment} account-id={data.accountId} account-name={data.accountName}></c-staff-availability-item>
+							<c-staff-availability-item namespace={namespace} assignment-record={assignment} account-id={data.accountId} account-name={data.accountName}></c-staff-availability-item>
```
Comment (cwico) [inline force-app/main/default/lwc/staffAvailability/staffAvailability.html:53]: very cool, databinding

File: force-app/main/default/lwc/appModal/appModal.js [hunk 8]
```diff
@@ -24,6 +24,9 @@ export default class AppModal extends LightningElement {
 	@api
 	modalSize;
 
+	@api
+	hideCloseButton = false;
```
Comment (cwico) [inline force-app/main/default/lwc/appModal/appModal.js]: "hide" or "show"  prefixes are more like action names, may be acceptable

isCloseButtonHidden =  false;

**Review by cwico — CHANGES_REQUESTED** [review]
This change looks awesome, learned a lot. Just minor renames would make it excellent.

File: force-app/main/default/lwc/staffAvailability/staffAvailability.js (hunk 2 above)
Comment (pserquiz) [reply force-app/main/default/lwc/staffAvailability/staffAvailability.js]: good catch, changing it now ^^

File: force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js (hunk 6 above)
Comment (pserquiz) [reply force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js:43]: yep, and make sure it changes the data both ways parent / child lwc so it rerenders when the values changes.

File: force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js (hunk 5 above)
Comment (pserquiz) [reply force-app/main/default/lwc/staffAvailabilityItem/staffAvailabilityItem.js:39]: could be anything really, just needed a get to create the set piece ^^

**Review by cwico — APPROVED** [review]
Yay

**Pull request merged**